modules/top-level/output.nix

```diff
@@ -285,12 +285,19 @@ in
         '';
       };
 
-      extraConfigLuaPre = lib.mkIf config.wrapRc ''
-        -- Ignore the user lua configuration
-        vim.opt.runtimepath:remove(vim.fn.stdpath('config'))              -- ~/.config/nvim
-        vim.opt.runtimepath:remove(vim.fn.stdpath('config') .. "/after")  -- ~/.config/nvim/after
-        vim.opt.runtimepath:remove(vim.fn.stdpath('data') .. "/site")     -- ~/.local/share/nvim/site
-      '';
+      extraConfigLuaPre = lib.mkOrder 1 (
+        # Add a global table at start of init
+        ''
+          local M = {}
+
+        ''
+        + lib.optionalString config.wrapRc ''
```

Comment:
Nit: this could lead to inconsistent line breaks.

If `wrapRc` is false you'll end up with `"\n\n\n"`; the third one coming from `concatLines` used in `types.lines` merge function.